### PR TITLE
References email changes

### PR DESCRIPTION
**Pull request by johndoates**
## Context

Some of the candidate emails about references were not the most recent versions. We need to update them before the new references journey is launched on 11 October.

## Changes proposed in this pull request

I've updated:

- conditions_met.text.erb
- offer_accepted.text.erb
- reinstated_offer.text.erb
- unconditional_offer_accepted.text.erb

I've fixed typos in:

- new_referee_request.text.erb
- chase_reference_again.text.erb

### Changed logic

These are all content changes, except that I removed logic from reinstated_offer.text.erb. 

Previously it distinguished between the old and new reference journey. This affected how the link to the service was shown. I moved the link so it only displays if the candidate has conditions.

I could have nested the old/new journey logic inside the conditions/no conditions logic. But that seemed unnecessary given that no users should see the old journey any more, and it also isn't a significant change.

### reinstated_offer in support

The current reinstated_offer email does not display correctly in support when there are no conditions. The lead in line for conditions is shown, but with no bullet list as there are no conditions. 

To me it looks like the logic is OK in the code (and I haven't changed it). Whoever reviews this should check whether we need to make a change either as a commit in this PR or as a separate PR.

CleanShot 2022-10-10 at 09 22 48

### Tests

I included commits to fix the tests broken by the subject line changes.

## Link to Trello card

[
](https://trello.com/c/5dMY5JDf/750-update-candidate-emails)

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/views/candidate_mailer/_new_flow_chase_reference_again.text.erb
Comment (frankieroberto) [inline app/views/candidate_mailer/_new_flow_chase_reference_again.text.erb:10]: Good spot!

File: app/views/candidate_mailer/conditions_met.text.erb
Comment (frankieroberto) [inline app/views/candidate_mailer/conditions_met.text.erb:12]: Is this meant to be a subheading? Think it needs two `##` and a space?
```suggestion
## If you’re unable to start training in <%= @start_date %>
```

Comment (MylesJarvis) [inline app/views/candidate_mailer/conditions_met.text.erb:12]: @johndoates do we know if start date is a Month and Year or a Day Month and Year? If it's the latter, does this need to be 'on' start date?

File: app/views/candidate_mailer/conditions_met.text.erb
Comment (frankieroberto) [inline app/views/candidate_mailer/conditions_met.text.erb:16]: Minor code style thing.
```suggestion
  Contact <%= @provider_name %> to find out if they’ll allow you to defer your offer. Asking if it’s possible will not affect your existing offer.
```

File: app/views/candidate_mailer/reinstated_offer.text.erb
Comment (frankieroberto) [inline app/views/candidate_mailer/reinstated_offer.text.erb:15]: Think this `else` and the following `end` need to go, as you’ve removed the `if` statement?

File: app/views/candidate_mailer/unconditional_offer_accepted.text.erb
Comment (frankieroberto) [inline app/views/candidate_mailer/unconditional_offer_accepted.text.erb:13]: Subheading?
```suggestion
## If you’re unable to start training in <%= @start_date %>
```

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Thanks for reviewing these!

I'm not 100% sure, but I think the leading whitespace (indentation) in all the emails might need to be removed to avoid Notify treating this as markdown syntax for fixed-width text? Might be wrong though!

File: app/views/candidate_mailer/reinstated_offer.text.erb
Comment (frankieroberto) [reply app/views/candidate_mailer/reinstated_offer.text.erb:15]: Oh actually I misread this - the `if` has gone, but so has the previous `else` and `end`, so this might be ok?

File: app/views/candidate_mailer/conditions_met.text.erb
Comment (johndoates) [reply app/views/candidate_mailer/conditions_met.text.erb:12]: It's the same variable as the one used in app/views/candidate_mailer/unconditional_offer_accepted.text.erb. So it should show just month and year, as in this screenshot.

CleanShot 2022-10-10 at 12 33 16@2x

Comment (johndoates) [reply app/views/candidate_mailer/conditions_met.text.erb:12]: I used the same syntax as app/views/candidate_mailer/unconditional_offer_accepted.text.erb, which currently renders correctly as a heading.

File: app/views/candidate_mailer/unconditional_offer_accepted.text.erb
Comment (johndoates) [reply app/views/candidate_mailer/unconditional_offer_accepted.text.erb:13]: I used the same syntax as app/views/candidate_mailer/unconditional_offer_accepted.text.erb, which currently renders correctly as a heading.

**Comment by johndoates** [comment]
> Thanks for reviewing these!
> 
> I'm not 100% sure, but I think the leading whitespace (indentation) in all the emails might need to be removed to avoid Notify treating this as markdown syntax for fixed-width text? Might be wrong though!

I haven't changed this. We could try sending the new emails in QA once the PR has been merged, to check that Notify does not use fixed-width text.

File: app/views/candidate_mailer/reinstated_offer.text.erb
Comment (johndoates) [reply app/views/candidate_mailer/reinstated_offer.text.erb:15]: I think it's OK, as long as we're fine to remove the logic giving different content for the old and new reference processes.

**Comment by johndoates** [comment]
There's a broken test in spec/mailers/candidate_mailer_spec.rb. On line 417 it expects content about references in the reinstated_offer email. But we've removed content about references from that email.

I think we should remove the test.

File: app/views/candidate_mailer/reinstated_offer.text.erb
Comment (MylesJarvis) [reply app/views/candidate_mailer/reinstated_offer.text.erb:15]: @JR-G this might need your eyes over it too if poss.

File: app/views/candidate_mailer/unconditional_offer_accepted.text.erb
Comment (MylesJarvis) [reply app/views/candidate_mailer/unconditional_offer_accepted.text.erb:13]: This feels like we should just resolve/not do anything.

File: app/views/candidate_mailer/conditions_met.text.erb
Comment (frankieroberto) [reply app/views/candidate_mailer/conditions_met.text.erb:12]: Interesting! Looks like notify does its own thing with markdown syntax...

File: app/views/candidate_mailer/reinstated_offer.text.erb
Comment (frankieroberto) [reply app/views/candidate_mailer/reinstated_offer.text.erb:15]: Sounds fine to me, as we're past the point of needing the old references flow 👍

File: app/views/candidate_mailer/unconditional_offer_accepted.text.erb
Comment (frankieroberto) [reply app/views/candidate_mailer/unconditional_offer_accepted.text.erb:13]: You're right, I'm wrong. 🤦

File: spec/mailers/candidate_mailer_spec.rb
Comment (JR-G) [reply spec/mailers/candidate_mailer_spec.rb:416]: Just had a look – i believe @johndoates removed it so this test no longer applies

Comment (johndoates) [reply spec/mailers/candidate_mailer_spec.rb:416]: Yes this is the test I mentioned in a comment above. I wasn't sure how much of the code to remove so I thought I'd leave it to someone else.

Comment (carlosmartinez) [reply spec/mailers/candidate_mailer_spec.rb:416]: will do so now

**Review by JR-G — APPROVED** [review]

**Pull request merged**
